### ISSUE TITLE
Supervised loss

### ISSUE THREAD

**Issue by hieuvt29** [jsyoon0823/TimeGAN#62]
Hi @jsyoon0823, 
Thank you for your great work and for sharing your code. 

I have a few questions related to the implementation of supervised loss introduced in the paper.
First, in Eq. 9:
![image](https://user-images.githubusercontent.com/18468247/203472991-905a41b1-746b-489c-a67c-53845389da84.png)
the generator takes in `z_t` as the input, while in the code, I see no `z` in the arguments here https://github.com/jsyoon0823/TimeGAN/blob/master/timegan.py#L132. 

Second, I see you create another `supervisor` network for that purpose instead of using the generator, and the loss here https://github.com/jsyoon0823/TimeGAN/blob/master/timegan.py#L200 (between `H=embedder(X, T)` and `H_hat_supervise=supervisor(H, T)`) has no connection with the generator, so I wonder how the supervision loss helps training generator (same question with issue #35). 

Third, what is the motivation for using `num_layers-1` layers in the supervisor network? (https://github.com/jsyoon0823/TimeGAN/blob/master/timegan.py#L143)

Please correct me if I misunderstand something.

Best, 
Hieu

**Comment by jsyoon0823** [comment]
Thank you for your interest in our paper.

1. Z in the generator
- https://github.com/jsyoon0823/TimeGAN/blob/master/timegan.py#L116

2. Supervisor
- Supervisor utilizes the output of the generator (
https://github.com/jsyoon0823/TimeGAN/blob/master/timegan.py#L170).
- You can think the combination of supervisor and generator is the final
generator.

3. num_layers - 1
- We have one more RNN layer here (
https://github.com/jsyoon0823/TimeGAN/blob/master/timegan.py#L144).
- Thus, we set it as num_layers - 1.

Thanks

**Comment by hieuvt29** [comment]
Thank you for your explanation, I get the point for the `generator + supervisor`, but for 3), the link you included is for `tf.nn.dynamic_rnn` function, and I believe it doesn't add extra layers to the network based on this [document ](https://www.tensorflow.org/api_docs/python/tf/compat/v1/nn/dynamic_rnn), where the architecture is specified by `e_cell` only, so I think I also get that one, there is no special reason for that `num_layers - 1`, I thought it was used for splitting the architecture somehow. 
Thanks!

**Comment by jsyoon0823** [comment]
Agreed. (3) is the mistake. Yes. We can use num_layers instead of
num_layers -1.
